src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string | null,
+  callback: Function,
+}
+
+export class App extends React.Component<{}, State>{
+  state = {
+    key: null,
+    callback: (event: KeyboardEvent) => {
```

Comment:
we pass to state data that can be changed. Callback function never changes so you don't need it in state